### PR TITLE
tests: collect the autopkgtest results and save them in git 

### PR DESCRIPTION
**Pull request by elopio**
- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] If this is a bugfix. Have you checked that there is a bug report open for the issue you are trying to fix on [bug reports](https://bugs.launchpad.net/snapcraft)?
- [x] If this is a new feature. Have you discussed the design on the [forum](https://forum.snapcraft.io)?
- [x] Have you successfully run `./runtests.sh static`?
- [x] Have you successfully run `./runtests.sh unit`?

-----

### CODE REVIEW

File: tools/collect_ppa_autopkgtests_results.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    subprocess.check_call([
+        'git', '-C', repo_dir, 'add' , '--all'])
+    subprocess.check_call([
+        'git', '-C', repo_dir, 'commit' , '--message',
+        'Add the results for {}'.format(day)])
+
+    subprocess.check_call([
+        'git', '-C', repo_dir, 'push',
+        'https://{GH_TOKEN}@github.com/elopio/autopkgtest-results.git'.format(
```
Comment (kyrofa) [inline tools/collect_ppa_autopkgtests_results.py]: Huh, I didn't realize the token was also used as the username.

File: tools/travis/run_ppa_autopkgtests.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 #
-# Copyright (C) 2015-2017 Canonical Ltd
+# Copyright (C) 2017 Canonical Ltd
```
Comment (kyrofa) [inline tools/travis/run_ppa_autopkgtests.sh]: Is this change on purpose?

File: tools/collect_ppa_autopkgtests_results.py (hunk 1 above)
Comment (kalikiana) [reply tools/collect_ppa_autopkgtests_results.py]: Official docs say it can be used in place of the password. Presumably you can just set it to foo:bar to use it the way it's done here. But perhaps we should document somewhere how we use it?

File: tools/travis/run_ppa_autopkgtests.sh (hunk 2 above)
Comment (elopio) [reply tools/travis/run_ppa_autopkgtests.sh]: Yes, the 2015 was a copy and paste mistake. This is a file we added this year.

File: tools/collect_ppa_autopkgtests_results.py (hunk 1 above)
Comment (elopio) [reply tools/collect_ppa_autopkgtests_results.py]: I'm not sure what are you suggesting to document. This is the common usage pattern for the token in scripts.

File: .travis.yml [hunk 3]
```diff
@@ -63,5 +65,9 @@ jobs:
       script: sudo ./tools/travis/run_autopkgtests.sh ubuntu-daily:bionic integrationtests-plugins-catkin
     - if: type = cron
       script: sudo ./tools/travis/run_autopkgtests.sh ubuntu-daily:bionic integrationtests-snapd
+    # Trigger autopkgtests in all the supported distros and architectures for the snapcraft-daily PPA
     - if: type = cron
-      script: ./tools/travis/run_ppa_autopkgtests.sh
+      script: sudo ./tools/travis/run_ppa_autopkgtests.sh
+    # Collect the results for the PPA autopkgtests that were triggered the day before.
+    - if: type = cron
+      script: sudo ./tools/travis/collect_ppa_autopkgtests_results.sh $(date -d "yesterday" "+%Y%m%d")
```
Comment (sergiusens) [inline .travis.yml]: sometimes tests take more than a day to run on other architectures, we should be a bit more smart here.
